src/orange/surf/Involute.hh

```diff
@@ -0,0 +1,347 @@
+//----------------------------------*-C++-*----------------------------------//
+// Copyright 2023-2024 UT-Battelle, LLC, and other Celeritas developers.
+// See the top-level COPYRIGHT file for details.
+// SPDX-License-Identifier: (Apache-2.0 OR MIT)
+//---------------------------------------------------------------------------//
+//! \file orange/surf/Involute.hh
+//---------------------------------------------------------------------------//
+#pragma once
+
+#include <cmath>
+
+#include "corecel/Types.hh"
+#include "corecel/cont/Array.hh"
+#include "corecel/cont/Span.hh"
+#include "corecel/math/ArrayUtils.hh"
+#include "orange/OrangeTypes.hh"
+
+#include "detail/InvoluteSolver.hh"
+
+namespace celeritas
+{
+//---------------------------------------------------------------------------//
+/*!
+ * Involute:
+ *
+ * \f[
+    x = r_b * (cos(t+a) + tsin(t+a)) + x_0
+    y = r_b * (sin(t+a) - tcos(t+a)) + y_0
+   \f]
+
+ * where \em t is the normal angle of the tangent to the circle of involute
+ with
+ * radius \em r_b from a starting angle of \em a (\f$r/h\f$ for a finite cone.
+ */
+class Involute
+{
+  public:
+    //@{
+    //! \name Type aliases
+    using Intersections = Array<real_type, 3>;
+    using StorageSpan = Span<real_type const, 4>;
```

Comment:
By my count, this class has 8 reals, not 4, so reconstructing this surface from type-deleted data will fail.